### PR TITLE
Handle two CR events by suppression approach

### PR DESCRIPTION
**Pull request by gkGaneshR**
Signed-off-by: gkGaneshR <gkganesh126@gmail.com>

**What this PR does / why we need it**:
When two consecutive events are received, sidecar cannot completely ignore
the second event, instead can prevent from getting propagated to CRs. However
pod log will continue to have the error.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*:
 fixes openebs/openebs#1730

### CODE REVIEW

**Comment by gkGaneshR** [comment]
This will be further handled by Moteesh cc: vishnu.attur

File: cmd/cstor-pool-mgmt/controller/pool-controller/handler.go [hunk 1]
```diff
@@ -48,17 +51,36 @@ func (c *CStorPoolController) syncHandler(key string, operation common.QueueOper
...
+
+	statusBytes, err := json.Marshal(cStorPoolGot.Status)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	finalizerBytes, err := json.Marshal(cStorPoolGot.Finalizers)
+	if err != nil {
+		log.Fatalln(err)
+	}
 	if err != nil {
```
Comment (princerachit) [inline cmd/cstor-pool-mgmt/controller/pool-controller/handler.go]: This line would become redundant after above check.

File: cmd/cstor-pool-mgmt/controller/pool-controller/handler.go [hunk 2]
```diff
@@ -48,17 +51,36 @@ func (c *CStorPoolController) syncHandler(key string, operation common.QueueOper
 		return nil
 	}
 	cStorPoolGot.Status.Phase = apis.CStorPoolPhase(status)
+
```
Comment (avishnu) [inline cmd/cstor-pool-mgmt/controller/pool-controller/handler.go]: Please add comments explaining the logic.

Comment (avishnu) [inline cmd/cstor-pool-mgmt/controller/pool-controller/handler.go]: Also, the previous error handling for this above line of code is missing with the latest changes.

**Comment by princerachit** [comment]
Looks like the code is trying to suppress two delete events only for cstor pool and not for cstor volume replica.

**Comment by avishnu** [comment]
Closing this PR as the functionality has been achieved in PR #467 .

**Pull request closed without merging**